libraries/abstractions/platform/freertos/iot_network_freertos.c

```diff
@@ -643,7 +648,8 @@ IotNetworkError_t IotNetworkAfr_Close( void * pConnection )
     ( void ) xEventGroupSetBits( ( EventGroupHandle_t ) &( pNetworkConnection->connectionFlags ),
                                  _FLAG_SHUTDOWN );
 
-    if( pNetworkConnection->receiveTask != NULL )
+    /* If this function is not called from the receive task, wait for the receive task to exit. */
+    if( ( pNetworkConnection->receiveTask != NULL ) && ( xTaskGetCurrentTaskHandle() == pNetworkConnection->receiveTask ) )
```

Comment:
Do we want to enter this block if this function is called from the receive task? The comment seems to indicate otherwise.

Reply:
Good catch, we don't want to enter this block! Have fixed it.